R/tree.R

```diff
@@ -0,0 +1,314 @@
+#' Pretty tree-like object printing
+#'
+#' A cleaner and easier to read replacement for `str` for nested list-like
+#' objects
+#'
+#' @param x A tree like object (list, etc.)
+#' @param index_arraylike Should children of containers without names have
+#'   indices used as stand-in?
+#' @param max_depth How far down the tree structure should be printed. E.g. `1`
+#'   means only direct children of the root element will be shown. Useful for
+#'   very deep lists.
+#' @param val_printer Function that values get passed to before being drawn to
+#'   screen. Can be used to color or generally style output.
+#' @param class_printer Same as `val_printer` but for the the class types of
+#'   non-atomic tree elements.
+#' @param show_attributes Should attributes be printed as a child of the list or
+#'   avoided?
+#' @param remove_newlines Should character strings with newlines in them have the
+#'   newlines removed? Not doing so will mess up the vertical flow of the tree
+#'   but may be desired for some use-cases if newline structure is important to
+#'   understanding object state.
+#' @param
+#' char_vertical,char_horizontal,char_branch,char_final_branch,char_vertical_attr,char_horizontal_attr
+#' Unicode characters used to construct the tree. Typically you wont want to
+#' change these.
+#' @param ... Ignored (used to force use of names)
+#'
+#' @return console output of structure
+#'
+#' @examples
+#'
+#' x <- list(
+#' list(id = "a",
```

Comment:
Could you please indent using standard tidyverse rules? https://style.tidyverse.org/syntax.html#long-lines

Reply:
Sorry about that! I was attempting to get the indenting of the object constructing code to match how it would print. I _think_ I fixed all the cases where I broke with the guidelines.